doc/tracecontext.md

```diff
@@ -0,0 +1,125 @@
+For an inbound request, the value from the request is applied to this same property.
+
+Tracing using the `TraceSource` API will include the value from `Trace.CorrelationManager.ActivityId` in each traced message if the [SourceLevels.ActivityTracing][SourceLevelsActivityTracingFlag] flag is set.
+Note that this flag *not* included in `SourceLevels.Verbose`.
+
+For example, to construct a `TraceSource` that will emit activity IDs for correlation between processes and machines, use code such as:
+
+```cs
+var traceSource = new TraceSource("some name", SourceLevels.Warning | SourceLevels.ActivityTracing);
+```
+
+Or to modify an existing `TraceSource` to add activity tracing, you can set the flag:
+
+```cs
+traceSource.Switch.Level |= SourceLevels.ActivityTracing;
+```
+
+You *may* share a `TraceSource` or `TraceListener` between your `JsonRpc` instance and the `CorrelationManagerTracingStrategy` instance.
+For proper activity recording in trace logs, be sure to set the `SourceLevels.ActivityTracing` flag described above in all relevant `TraceSource` objects.
+
+### Creating and reviewing trace logs
+
+Reviewing trace files is much easier, particularly when reviewing many files that may span processes and/or machines, when using a tool such as [Service Trace Viewer][ServiceTraceViewer].
+This viewer reads trace files written with the [`XmlWriterTraceListener`](XmlWriterTraceListener).
+Add an instance of this class to your `TraceSource.Listeners` collection for the best trace log viewing experience.
+This can be used in combination with other unstructured `TraceListener`-derived classes if desired.
+
+## Protocol
+
+The formatting, parsing, propagating and modification rules all apply as defined in the [Trace Context][trace-context] spec.
+
+The `tracestate` property is optional and MAY be omitted when empty.
+The `traceparent` property is optional but MUST be present if `tracestate` is present.
+
+### Text encoding
+
+When using a text-based encoding (e.g. UTF-8) the trace-context values are encoded as strings as follows:
+
+Given a standard JSON-RPC request:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "pick",
+  "params": [],
+  "id": 1
+}
+```
+
+We include the trace context by adding a property to the JSON-RPC message for each of the HTTP headers described in the [Trace Context W3C spec][trace-context]: [`traceparent`][traceparent] and [`tracestate`][tracestate].
+
+For example, a JSON-RPC request message may look like this:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "pick",
+  "params": [],
+  "id": 1,
+  "traceparent": "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01",
+  "tracestate": "rojo=00f067aa0ba902b7,congo=t61rcWkgMzE"
+}
+```
+
+### Binary encoding
+
+When using a binary encoding (e.g. MessagePack) the trace-context values are encoded as arrays of binary elements as follows:
+
+- `traceparent` as `[uint8, [byte[], byte[], byte]]`
```

Comment:
I've changed `byte` to `uint8` above.
The inner array costs 1 byte. So ya, it's just a convenient wrapper so that depending on the version, the serialization process can engage another formatter/deformatter that can expect to deserialize/serialize an entire 'object'.